### ISSUE TITLE
Tensorflow conv1d computed incorrectly on GPU

### ISSUE THREAD

**Issue by edanh88** [tensorflow/tensorflow#53533]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: 
- TensorFlow installed from (source or binary):  binary
- TensorFlow version (use command below): 2.7
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: Cuda-11.2
- GPU model and memory: Tesla K80

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
Computing a single filter 1d convolution on GPU with kernel_size=15, padding='valid' and strides=1 on a (1, 64, 32) input tensor (1 sample X 64 coordinates X 32 input channels) results in data from the end of the input tensor effecting the beginning of the output vector. 

This behavior only happens when performing the calculation on GPU (specifically nvidia Telsa K80). 
On CPU results are fine.

**Describe the expected behavior**
Receptive field should be limited to the kernel size. 

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): no
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Issue reproduces on colab with GPU:
https://colab.research.google.com/drive/1pK1tNxWVtC9qilcX0u5sbpGmQEZiCDEX?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
[Tensorflow-bug.ipynb - Colaboratory.pdf](https://github.com/tensorflow/tensorflow/files/7769397/Tensorflow-bug.ipynb.-.Colaboratory.pdf)

**Comment by mohantym** [comment]
Hi @edanh88 ! I changed the runtime environment to [CPU](https://colab.sandbox.google.com/gist/mohantym/d4faa2bdc668b85261732fc18d8015e5/tensorflow-bug.ipynb#scrollTo=ZyXg6Bdgv1OP) and then to [GPU](https://colab.sandbox.google.com/gist/mohantym/d9330230394869e3e2c19c6287c9cb52/tensorflow-bug.ipynb#scrollTo=cnL4334XwalE)  and ran the the above [code](https://colab.sandbox.google.com/gist/mohantym/fec5233cc88b49ef5ffcb88ba3ee0afa/tensorflow-bug.ipynb#scrollTo=L79VLeg0rNcV) along. Issue was replicating with original code  in first attempt though. Thanks!

**Comment by chunduriv** [comment]
@edanh88, 

Did you found the same using different GPU cards?  Can you brief `On CPU results are fine`? Thanks!

**Comment by edanh88** [comment]
I didn't try different GPU cards. I've tried multiple instances of Nvidia Tesla K80 because that is the only card available to me at the moment.

when running on CPU (either by setting CUDA_VISIBLE_DEVICES="" ) or by changing runtime environment in Colab, or by using "with tf.device("CPU:0")" as can be seen in the Colab example there is no error in the calculation. 

When run on GPU, I simply get the wrong results (basically I'm convolving a kernel of tf.ones with inputs of tf.ones so I expect the resulting tensor to have the value of KERNEL_WIDTH*INPUT_CHANNELS - and this is **not** what I get on GPU).
Moreover, one can see that values outside of the convolution's receptive field affect the results of the convolution. 

This does not happen on CPU. 
On CPU the results are exactly as expected and do not depend on values outside of the receptive field.

The 7'th cell in the colab example contains a few print lines - they should all print 0 if there is no error. And indeed when run without GPU (by setting runtime environment) they do.

Thanks,
Edan

**Comment by edanh88** [comment]
Hi @sanjoy @sachinprasadhs,
Sorry to bother you.
Just wondering if this is being addressed in some way? 
Thanks,
Edan

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

**Comment by edanh88** [comment]
Thanks, but since you didn't solve the issue in a timely manner I have
since swiched to PyTorch. Good luck.
